src/components/LocationSelector.vue

```diff
@@ -300,12 +306,26 @@ export default defineComponent({
       });
       
       this.map = map;
+      
+      this.$emit('mapReady');
     },
 
     updateValue(value: LocationDeg) {
       this.$emit('update:modelValue', value);
     },
 
+    addLayerToMap<T extends L.Layer>(leafletObject: T) {
```

Comment:
We don't need this function to be generic since we don't actually do anything with the generic type - we can just make the signature be `addLayerToMap(leafletObject: L.Layer)`, since this will accept anything that extends `L.Layer` as well.